### ISSUE TITLE
Nexus 4 issue

### ISSUE THREAD

**Issue by highpowerxh** [ikarus23/MifareClassicTool#1]
The NFC hardware is working normal on my Nexus 4, But it pops up a dialog saying No Mifare Support onCreate.
I cloned this project and I'll check that out.

**Comment by ThomasCZ** [comment]
Samsung Galaxy S6 and S6 Edge have not support for Mifare Classic.

**Comment by ikarus23** [comment]
Added Samaung's Galaxy S6 and S6 Edge to the incompatibility list (d77b72e4bad8b0c1a77c6d9a22b9a392d03e76db).

**Comment by zanechua** [comment]
Motorola Moto X 2014 does not have support for Mifare Classic

**Comment by Ugion** [comment]
@zanechua Can confirm, same thing here.

**Comment by ikarus23** [comment]
Thanks for reporting. I added the Moto X (2014) to the incompatibility list (1c80a8b06d28b09242baeba78e890ba47b14b3e2).

**Comment by silentjet** [comment]
MotoX 1st gen (2013) works fine

**Comment by ngoding** [comment]
Samsung A5 2017 does not support Mifare Classic

**Comment by levoniust** [comment]
Super old thread, but is the Galaxy Note 8  SM-N950U  supported?

**Comment by ikarus23** [comment]
Looking at the [list of incompatible devices](https://github.com/ikarus23/MifareClassicTool/blob/master/INCOMPATIBLE_DEVICES.md) I don't think the Note 8 will work. I'm sorry.

**Comment by sara-caballero** [comment]
How can I find out which devices support MIFARE Classic? My phone is not listed among either the compatible or incompatible devices. My phone is Oppo Find X5 Pro

**Comment by ikarus23** [comment]
Have a look at https://github.com/ikarus23/MifareClassicTool/discussions/467. I responded there.

**Comment by lorn10** [comment]
Hi all, really sorry to warm up this old old topic. However, Mifare Classic support on more recent Andoid based phones is still a thing even in 2026. ~Whatever, I can confirm that it looks that also the Samsung XCover 5 phone model doesn't  support it.~ **Edit**: It turned out that the tested RFID key fobs were not of Mifare or NTAG type which was effectely the reason why those key fobs didn't work. 

~And this seems to be true for the majority of more recent Samsung phone models not older than perhaps 10 years.~ **Edit**: The main requirement regarding any Mifare support is the presence of an NXP NFC chip. I think it can be said that Mifare support is practically non-existent on most non-NXP NFC chipsets because it is a technology patented by NXP.

Therefore it would be really helpful if there could be added some sort of detection in the MifareClassicTool which shows right at the first start a big fat warning that Mifare Classic is simply not supported. It looks that this could be added via the `android.nfc.tech.MifareClassic` [function](https://developer.android.com/reference/android/nfc/tech/MifareClassic) in Java. 🤔 

```
// Inside your tag discovered intent
Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
for (String tech : tag.getTechList()) {
    if (tech.equals("android.nfc.tech.MifareClassic")) {
        // Device supports MIFARE Classic
    }
}
```
So this would effectively allow end-users to use MifareClassicTool as a Mifare support detection app. 👍

**Comment by ikarus23** [comment]
Hi @lorn10. First of all: Thanks for reporting.

I'm a bit surprised by your assessment that most Samsung phones of the recent years have no MIFARE Classic support. I'm pretty sure they have. I see a lot of MCT installations on Samsung devices.

Regarding the check: MCT initially had a check at startup to find out if the device supports MIFARE Classic. However, it had to be removed because of false-positives. Some manufacturers claim compatibility but have none and vice versa. So I had to change the detection. This is a full rabbit hole. If you want to learn more, look at the comments and code in.

- https://github.com/ikarus23/MifareClassicTool/blob/41be4c697ffbf0a24050ed31ab1d68474941ddb3/Mifare%20Classic%20Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java#L808
- https://github.com/ikarus23/MifareClassicTool/blob/41be4c697ffbf0a24050ed31ab1d68474941ddb3/Mifare%20Classic%20Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java#L894
- https://github.com/ikarus23/MifareClassicTool/blob/41be4c697ffbf0a24050ed31ab1d68474941ddb3/Mifare%20Classic%20Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java#L751

In short: There is no good detection because so many manufacturer are not compiling to the specification. I've talked to other NFC app creators. They were facing the same issue and wrote their own detection.

Hope this helps. Cya.

**Comment by lorn10** [comment]
Thanks for the fast and informative reply. 👍 

My assumption regarding the MIFARE Classic support on newer Samsung devices could be also wrong. Perhaps the statement that the NFC chip must be made by NXP would be more accurate. 

In my case, my last phone which supported MIFARE Classic was the [Samsung Galaxy Mini 2](https://en.wikipedia.org/wiki/Samsung_Galaxy_Mini_2) from 2012 which came by the way with an NXP PN544 NFC Chip. All later models didn't were compliant. Perhaps this was just bad luck. Or all of my later phones simply didn't came with an NXP based NFC chip which would explain why MIFARE support was lacking.

Whatever, today I invested quite some time into the behavior of RFID key fobs on my XCover 5. My conclusion was that [MIFARE Classic](https://www.nfc-tag-shop.de/en/NFC-tag-ABS-40-x-32-mm-MIFARE-Classic-EV1-1k-1024-Byte-grey/68211) based ones are all non-compatible while [NTAG](https://www.nfc-tag-shop.de/NFC-Anhaenger-ABS-40-x-32-mm-NTAG216-924-Byte-grau/17207) based key fobs are working usually.

So yeah, this RFID topic is complicated in itself. If you want to be on the safe side, then just use NTAG based hardware which is an open standard while MIFARE is proprietary NXP tech. 😉

**Comment by lorn10** [comment]
Here follows a short addition / correction. 

I bought some new Mifare Classic 1k key fobs and those are working with my Samsung XCover 5. So the XCover 5 definitely supports Mifare based tags. I was wrong in that regard because the originally tested key fobs were not of MIFARE or NTAG type which are sending both at [13.56 MHz](https://en.wikipedia.org/wiki/Near-field_communication). Although the tested key fobs are looking from the outside identical, those are most likely of a different variant which is operating at a (lower) frequency. That would explain also why those "RFID key fobs" are not detected by any phone.

As far I understand NFC capable phones are exclusively operating at 13.56 MHz. Other RFID frequencies are almost never supported. So I have learned here that the term "RFID" is in the first place a collective term which can describe various forms of radio-frequency identification technologies which can be quite different from each other.

**Comment by ikarus23** [comment]
Thanks for the update. And yes, you learned a powerful lesson about RFID and NFC. :) You are correct, NFC are a collection of RFID standards around 13.56 MHz. RFID is a broader term for a lot of things related to identification via radio frequencies. But even in NFC there are big difference ([this picture](https://en.wikipedia.org/wiki/Near-field_communication#/media/File:NFC_Protocol_Stack.png) gives some examples).

If you want a quick tip how you might identify 13.56 MHz vs. 125 kHz: Shine a bright light through (easier for cards than key fobs). 125 kHz coils have way more winds. Here are some examples:

- 125 kHz: https://doegox.github.io/rfidpics/#!/see-through_lf
- 13.56 MHz: https://doegox.github.io/rfidpics/#!/see-through_hf

We're getting off topic. Anyways, good luck and have fund in the world of NFC/RFID
